modules/biobank/php/container.class.inc

```diff
@@ -0,0 +1,339 @@
+<?php declare(strict_types=1);
+/**
+ * Container Object Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Container Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\biobank;
+
+/**
+ * Container Object Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Container Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category Loris
+ * @package  Biobank
+ * @author   Henri Rabalais <hrabalais.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+class Container implements \JsonSerializable, \LORIS\Data\DataInstance
+{
+    /**
+     * Persistent Instance variables.
+     *
+     * These variables are directly mapped to the fields of the database schema
+     * that describe a container entity.
+     *
+     * @access private
+     */
+    private $id;                 //int
+    private $barcode;            //int
+    private $typeId;             //int
+    private $dimensionId;        //int
+    private $temperature;        //float
+    private $statusId;           //int
+    private $originId;           //int
+    private $centerId;           //int
+    private $parentContainerId;  //int
+    private $childContainerIds;  //array
+    private $coordinate;         //int
+
+    /**
+     * Initiliazes a new instance of the Container Class
+     */
+    function __construct()
```

Comment:
Maybe this should have setters inside? It could help you cut down on your checks for typing and null values below if you declare a specimen with all the required fields here.

Reply:
That's definitely a possibility. I think this design question will definitely have to be addressed as we move to standardise `DataInstance`/`Entity` classes. Since this would a decent amount of code change to implement, I'm going to leave this as is for now though and see if anyone else has any opinions.

@ridz1208 @xlecours @driusan

Reply:
this will come up at the design meeting. TBD